### PR TITLE
Display "To be confirmed*" for inquiry make offer works with missing tax and shipping price

### PR DESCRIPTION
**Pull request by lilyfromseattle**
# [PURCHASE-2598]

Paired with @xtina-starr 

This change updates the tax and shipping price copy on the Review page from a dash to To be confirmed* for inquiry offers with missing metadata, and adds a brief message about gallery confirmation

As per [this discussion](https://artsy.slack.com/archives/C9YNS4X32/p1619455302136600), this change is only for the Review page and only for inquiry checkout orders with missing metadata.

In this PR we check for the presence of `domesticShippingFee` and `internationalShippingFee` to display the special message. We considered also checking for `shippingOrigin` but decided against when we saw that the test artwork @starsirius created had a `shippingOrigin`, but we can change this if that was a mistake.

NOTE: As part of post review changes, we updated the logic to rely on the new `hasDefiniteTotal` field instead of the variosu shipping and tax fields

We originally pulled this ticket: [PURCHASE-2563], but came to the conclusion that it was best to make this change client side, so we separated out the CMS and force work.

[PURCHASE-2598]: https://artsyproduct.atlassian.net/browse/PURCHASE-2598
[PURCHASE-2563]: https://artsyproduct.atlassian.net/browse/PURCHASE-2563

### CODE REVIEW

File: src/v2/Apps/Order/Routes/Review/index.tsx [hunk 1]
```diff
@@ -318,6 +319,10 @@ export class ReviewRoute extends Component<ReviewProps> {
 
   render() {
     const { order, isCommittingMutation } = this.props
+    const { artwork } = order.lineItems.edges[0].node
+
+    const isMissingMetadata =
+      !artwork.domesticShippingFee && !artwork.internationalShippingFee
```
Comment (xtina-starr) [inline src/v2/Apps/Order/Routes/Review/index.tsx:325]: This is where we previously had the check for `shippingOrigin`. The current implementation covers shipping fee however taxes is not considered here. Does shipping origin somehow contribute to calculating tax? Better question what property would we use to determine if the tax line item is available to display or not or not?

Comment (lilyfromseattle) [reply src/v2/Apps/Order/Routes/Review/index.tsx:325]: I added a check for `taxTotal` here

File: src/v2/Apps/Order/Components/TransactionDetailsSummaryItem.tsx [hunk 2]
```diff
@@ -96,7 +105,9 @@ export class TransactionDetailsSummaryItem extends React.Component<
       <>
         <Entry
           label={isBuyerOffer ? "Your offer" : "Seller's offer"}
-          value={offerOverride || (offer && offer.amount) || emDash}
+          value={
+            offerOverride || (offer && offer.amount) || this.amountPlaceholder
```
Comment (starsirius) [inline src/v2/Apps/Order/Components/TransactionDetailsSummaryItem.tsx:109]: Is the "to be confirmed" copy going to be rendered on the price field in some scenarios? I wonder if the review step would always have a price.

File: src/v2/Apps/Order/Routes/Review/index.tsx [hunk 3]
```diff
@@ -383,9 +388,19 @@ export class ReviewRoute extends Component<ReviewProps> {
...
+                    order={order}
+                    placeholderOverride={
+                      isMissingMetadata ? "To be confirmed*" : null
+                    }
+                  />
                 </Flex>
                 <BuyerGuarantee />
+                {isMissingMetadata && (
+                  <Text variant="small" color="black60">
+                    *Shipping and taxes to be confirmed by gallery
```
Comment (starsirius) [inline src/v2/Apps/Order/Routes/Review/index.tsx:401]: It's possible only one of shipping cost and tax is missing, so we can probably separate them out and check the presence of `shippingTotalCents` and `taxTotalCents` individually.

File: src/v2/Apps/Order/Routes/Review/index.tsx [hunk 4]
```diff
@@ -383,9 +388,19 @@ export class ReviewRoute extends Component<ReviewProps> {
               <Flex flexDirection="column">
                 <Flex flexDirection="column">
                   <ArtworkSummaryItem order={order} />
-                  <TransactionDetailsSummaryItem order={order} />
+                  <TransactionDetailsSummaryItem
+                    order={order}
+                    placeholderOverride={
+                      isMissingMetadata ? "To be confirmed*" : null
```
Comment (starsirius) [inline src/v2/Apps/Order/Routes/Review/index.tsx:394]: In this "review" step, we can probably just pass `placeholderOverride="To be confirmed*"` given it's the last step and if there is shipping or tax is still missing it is to be confirmed. Or do I miss anything here?

File: src/v2/Apps/Order/Components/TransactionDetailsSummaryItem.tsx [hunk 5]
```diff
@@ -17,18 +17,27 @@ export interface TransactionDetailsSummaryItemProps
...
-const emDash = "—"
-
 export class TransactionDetailsSummaryItem extends React.Component<
   TransactionDetailsSummaryItemProps
 > {
   static defaultProps: Partial<TransactionDetailsSummaryItemProps> = {
     offerContextPrice: "LIST_PRICE",
   }
+
+  amountPlaceholder = this.props.placeholderOverride || "—"
```
Comment (starsirius) [inline src/v2/Apps/Order/Components/TransactionDetailsSummaryItem.tsx:30]: Nice! 👍

File: src/v2/Apps/Order/Components/TransactionDetailsSummaryItem.tsx (hunk 2 above)
Comment (lilyfromseattle) [reply src/v2/Apps/Order/Components/TransactionDetailsSummaryItem.tsx:109]: The review step always has a price - if and only shipping and tax are available. So it should only render `to be confirmed` for inquiry offers with missing metadata.

File: src/v2/Apps/Order/Routes/Review/index.tsx (hunk 1 above)
Comment (lilyfromseattle) [reply src/v2/Apps/Order/Routes/Review/index.tsx:325]: Good catch, changed this to an `||`

**Comment by xtina-starr** [comment]
@starsirius did you have any addition feedback for this PR?

File: src/v2/Apps/Order/Routes/Review/index.tsx (hunk 1 above)
Comment (starsirius) [reply src/v2/Apps/Order/Routes/Review/index.tsx:325]: Now that we have `hasDefiniteTotal` defined in Exchange, we should switch to use that. Something like

```typescript
const isMissingMetadata = order.myLastOffer && !order.myLastOffer.hasDefiniteTotal
```

Probably need to update the relay query to get that field.

File: src/v2/Apps/Order/Routes/Review/index.tsx (hunk 3 above)
Comment (starsirius) [reply src/v2/Apps/Order/Routes/Review/index.tsx:401]: I have a [discussion](https://www.figma.com/file/JE8dGuL81Kqd06R4egxdZo?node-id=118%3A684&fuid=825389106351956176#79189851) with @ArvindLall to make the messaging generic, and we can iterate on this. 👍

**Comment by starsirius** [comment]
@xtina-starr @lilyfromseattle Final [comment](https://github.com/artsy/force/pull/7422/files#r628350909) on using the `hadDefiniteTotal` flag from Exchange directly; otherwise looks great to me!

File: src/v2/Apps/Order/Routes/Review/index.tsx (hunk 1 above)
Comment (xtina-starr) [reply src/v2/Apps/Order/Routes/Review/index.tsx:325]: I thought we made the switch to use the new flag. @lilyfromseattle did our latest changes get pushed?

File: src/v2/Apps/Order/Routes/Review/index.tsx [hunk 6]
```diff
@@ -376,9 +377,17 @@ export class ReviewRoute extends Component<ReviewProps> {
...
               <Flex flexDirection="column">
                 <ArtworkSummaryItem order={order} />
-                <TransactionDetailsSummaryItem order={order} />
+                <TransactionDetailsSummaryItem
+                  order={order}
+                  placeholderOverride="To be confirmed*"
+                />
               </Flex>
               <BuyerGuarantee />
+              {!order.myLastOffer?.hasDefiniteTotal && (
```
Comment (starsirius) [inline src/v2/Apps/Order/Routes/Review/index.tsx:386]: @lilyfromseattle @xtina-starr Sorry to lose track of this PR. Would this also render the message for Buy Now orders? Maybe we need something like this?
```typescript
order.myLastOffer && !order.myLastOffer.hasDefiniteTotal
```

File: src/v2/Apps/Order/Routes/Review/index.tsx [hunk 7]
```diff
@@ -334,7 +335,7 @@ export class ReviewRoute extends Component<ReviewProps> {
               <Join separator={<Spacer mb={3} />}>
                 <Flex flexDirection="column" mb={[2, 3]}>
                   <Message p={[2, 3]} mb={[2, 3]}>
-                    Disruptions caused by COVID-19 may cause delays — we
+                    Disruptions caused by COVID-19x may cause delays — we
```
Comment (starsirius) [inline src/v2/Apps/Order/Routes/Review/index.tsx:338]: Is this intentional?

**Pull request merged**